qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/{self.subscription_id}"
+            f"/resourceGroups/{self.resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{self.workspace_name}/jobs/{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.get(url, headers=header)
+        return response.json()
+
+    def get_job_data(self, job_id: str) -> dict[str, Any]:
+        """Get the data from a specific Azure Quantum job."""
+
+        data = self._helpers.get_output_data(f"{self.jobs[job_id][1]}", self.connection_string)
+        return data
+
+    def cancel_job(self, job_id: str):
+        """Cancel a specific Azure Quantum job."""
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/"
+            f"{self.subscription_id}/resourceGroups/{self.resource_group_name}/"
+            f"providers/Microsoft.Quantum/workspaces/{self.workspace_name}/jobs/"
+            f"{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.delete(url, headers=header)
+        print("Job has been cancelled.")
+        return str(response)
+
+
+class AzureQuantumProvider(QuantumProvider):
+    """Azure provider class."""
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        tenant_id: str,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        storage_account: str,
+        connection_string: str,
+    ):  # pylint: disable= too-many-arguments
+        super().__init__()
+        self.session = AzureSession(
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant_id=tenant_id,
+            location_name=location_name,
+            subscription_id=subscription_id,
+            resource_group_name=resource_group_name,
+            workspace_name=workspace_name,
+            storage_account=storage_account,
+            connection_string=connection_string,
+        )
+
+    def _build_profile(self, data: dict[str, Any]) -> TargetProfile:
+        """Build a profile for an Azure device."""
+
+        device = data.get("id")
+
+        if "qpu" not in device:
+            device_type = DeviceType.SIMULATOR
+        else:
+            device_type = DeviceType.QPU
+
+        if "rigetti" in device:
+            program_spec = ProgramSpec(pyquil.Program)
+        else:
+            program_spec = ProgramSpec(openqasm3.ast.Program)
+
+        queue_time = data.get("averageQueueTime")
+        status = data.get("status")
+
+        return TargetProfile(
+            device_id=data.get("id"),
+            device_type=device_type,
+            program_spec=program_spec,
+            queue_time=queue_time,
+            status=status,
```

Comment:
`TargetProfile` is meant for static values that describe the characteristics of a quantum device and the runtime configurations that dictate how and if a program can be submitted to that device.

Values such as `queue_time` and `status` are dynamic in nature. They might be different today then they are tomorrow. Therefore, we should retrieve these values via API request in real-time from inside the device class.